### PR TITLE
PW-1102 PW-1111 Hide BCMC_mobile_QR and wechatpay except web

### PR DESCRIPTION
**Pull request by cyattilakiss**

### CODE REVIEW

File: cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js [hunk 1]
```diff
@@ -117,11 +117,63 @@ function getPaymentMethods(cart) {
...
+function isPaymentMethodTypeNotFilteredOut(methodType) {
+	if (!isPaymentMethodTypeScheme(methodType) && 
+		!isPaymentMethodTypeBcmcMobileQR(methodType) &&
+		!isPaymentMethodTypeWechatpayExceptWeb(methodType)
+	) {
+		return true;
+	}
+	
+	return false;
+}
```
Comment (rkewlani) [inline cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js]: I think we can simplify this a bit more by adding a common method for filtered types.

File: cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js [hunk 2]
```diff
@@ -117,11 +117,63 @@ function getPaymentMethods(cart) {
...
+/**
+ * Checks if payment method is not filtered out
+ */
+function isPaymentMethodTypeNotFilteredOut(methodType) {
+	if (!isPaymentMethodTypeScheme(methodType) && 
+		!isPaymentMethodTypeBcmcMobileQR(methodType) &&
+		!isPaymentMethodTypeWechatpayExceptWeb(methodType)
+	) {
+		return true;
+	}
```
Comment (rkewlani) [inline cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js]: //check if method type is blocked
isMethodTypeClocked(String methodType)
{
               if (methodType.indexOf('bcmc_mobile_QR') !== -1) 
		return true;
if ((methodType.indexOf('wechatpay') !== -1 && ! (methodType.indexOf('wechatpayWeb') !== -1)) 
return true;
if (methodType == "scheme") 
return true;
}

and we can remove other functions too.

File: cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js (hunk 1 above)
Comment (maassenbas) [reply cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js]: Agree

File: cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js [hunk 3]
```diff
@@ -117,11 +117,26 @@ function getPaymentMethods(cart) {
     if (Site.getCurrent().getCustomPreferenceValue("Adyen_directoryLookup")) {
         var	getPaymentMethods = require('int_adyen_overlay/cartridge/scripts/adyenGetPaymentMethods');
         var paymentMethods = getPaymentMethods.getMethods(cart.object).paymentMethods;
-        return paymentMethods.filter(function (method) { return method.type != "scheme"; })
+        return paymentMethods.filter(function (method) { 
+        	return isPaymentMethodTypeNotFilteredOut(method.type);
```
Comment (maassenbas) [inline cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js]: Shouldn't this be "isMethodTypeBlocked(method.type)"?

Comment (maassenbas) [inline cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js]: Shouldn't this be "isMethodTypeBlocked(method.type)"?

**Pull request merged**